http/peer-id-auth.md

```diff
@@ -0,0 +1,194 @@
+| ---------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| hostname         | The server name used in the TLS connection (SNI).                                                                                                                                                                                        |
+| challenge-server | The random base64 encoded value the client generates to challenge the server to prove its identity                                                                                                                                       |
+| challenge-client | The random base64 encoded value the server generates to challenge the client to prove its identity                                                                                                                                       |
+| sig              | A base64 encoded signature.                                                                                                                                                                                                              |
+| peer-id          | The peer id of the node that set this parameter. Encoding defined by the [Peer ID spec].                                                                                                                                                 |
+| public-key       | A base64 encoded value of peer's public key. The key itself is encoded per the [Peer ID spec].                                                                                                                                           |
+| opaque           | An base64 encoded opaque to the client blob generated by the server. If a client receives this it must return it. A server may use this to authenticate statelessly. For example, it could store the challenge-client and a expiry time. |
+
+Params are encoded per [RFC 9110 auth-param's ABNF](https://datatracker.ietf.org/doc/html/rfc9110#name-collected-abnf). Generally it'll be something like: `hostname="example.com", challenge-server=base64EncodedVal`
+
+## Signing
+
+Signatures sign some set of parameters. The parameters are sorted
+alphabetically, prepended with a varint length prefix, and concatenated together
+to form the data to be signed. The signing algorithm is defined by the key type
+used. Refer to the [Peer ID
+spec] for
+specifics on the signing algorithm. The set of parameters is prefixed with the auth scheme "libp2p-PeerID"
+
+As an example, if we wanted to sign the parameters `hostname="example.com",
+challenge-client=base64String` we would first structure the parameters as a byte
+slice containing:
+```
+libp2p-PeerID<varintprefix>challenge-client=<base64String><varintprefix>hostname="example.com"
```

Comment:
```suggestion
libp2p-PeerID<varintprefix>challenge-client=<base64String><varintprefix>hostname=example.com
```

Why do we want to quote this? `challenge-client=<base64string>` doesn't have the base64 string quoted

Reply:
It's following the ABNF of what goes in these headers. Specifically this is an `auth-param` which can be a token or a quoted string. `auth-param = token BWS "=" BWS ( token / quoted-string )`. However, reviewing this again, I don't think the padding `=` of base64url makes it count as a token. So it needs to also be a quoted string.

Reply:
I've updated to quote all of the params.